### ISSUE TITLE
Swagger with Optional values

### ISSUE THREAD

**Issue by dlangerenken** [vojtechhabarta/typescript-generator#271]
I have the following attribute in my class:

```
    @ApiModelProperty(dataType = "java.lang.String")
    @JsonProperty(value = "id")
    @JsonDeserialize(using = IdDeserializer.class)
    public Optional<Long> id;
```
Unfortunately, it will be converted to `id: string`, whereas I was hoping it to be `id?: string`
Is there any chance to make this specific attribute optional?

Note: ApiModelProperty has a 'required' parameter - maybe you can read this?

**Comment by vojtechhabarta** [comment]
I think it is better not to combine information from field type with Swagger annotation. I would use all needed information either from type or from annotation. So when `dataType` is not specified field type would be used, otherwise `dataType` and `required` (as you suggested) would be used.

However it is potentially breaking change. But I feel it is better to change it sooner rather than later. I will do it.

**Comment by dlangerenken** [comment]
I agree with you that you shouldn't combine field type with swagger annotations. 'required' in the ApiModelProperty might be the best scenario here.

Thanks for the quick response! Let me know once you need someone to verify the implementation!

**Comment by vojtechhabarta** [comment]
Released [v2.7.447](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v2.7.447).
Could you please try it?

**Comment by dlangerenken** [comment]
Hey, I'm actually getting > No input classes found. error - will investigate and let you know!

**Comment by dlangerenken** [comment]
It seems like the classPatterns are no longer triggering correctly.
I'm using pattern such as 'company.package.subpackage.*' and it used to work in the previous release

**Comment by vojtechhabarta** [comment]
In version 2.7.447 FastClasspathScanner was updated to ClassGraph which might have broke some features based on classpath scanning.

Could you please confirm that v2.6.433 is working while v2.7.447 is not working?

**Comment by dlangerenken** [comment]
I can confirm that 2.6.433 was correctly picking up all my files - I guess the ClassGraph might have introduced some bug.

**Comment by vojtechhabarta** [comment]
This issue was released in [v2.7.447](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v2.7.447).
For problem with classpath scanning we can reopen issues #280 if needed.
